// @taraldefi/tariala-component-library/build/index.esm.js
import React, { forwardRef, useState, useEffect } from 'react';

/******************************************************************************
Copyright (c) Microsoft Corporation.

Permission to use, copy, modify, and/or distribute this software for any
purpose with or without fee is hereby granted.

THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
PERFORMANCE OF THIS SOFTWARE.
***************************************************************************** */

var __assign = function() {
    __assign = Object.assign || function __assign(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};

function __rest(s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
}

function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (!css || typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var css_248z$7 = ".button--primary {\n  display: flex;\n  border-radius: 5px;\n  border: none;\n  transition: background 150ms;\n  font-family: \"inter\", sans-serif;\n  color: #ffffff;\n  cursor: pointer;\n}\n.button--primary span {\n  font-weight: 500;\n}\n\n.button--container {\n  gap: 5px;\n  display: flex;\n  align-items: center;\n  justify-content: center;\n}\n\n.button--secondary {\n  font-family: \"inter\", sans-serif;\n  display: flex;\n  border-radius: 5px;\n  border: #0d8489 solid 1px;\n  transition: background 150ms;\n  background: #ffffff;\n  cursor: pointer;\n  color: #1ab98b;\n}\n\n.button--small {\n  font-size: 12px;\n  padding: 10px 16px;\n}\n\n.button--medium {\n  font-size: 12px;\n  padding: 10px 20px;\n}\n\n.button--large {\n  font-size: 16px;\n  padding: 12px 24px;\n}";
styleInject(css_248z$7);

var Button = function (_a) {
    var _b = _a.primary, primary = _b === void 0 ? false : _b, _c = _a.size, size = _c === void 0 ? 'medium' : _c, backgroundColor = _a.backgroundColor, label = _a.label, icon = _a.icon, props = __rest(_a, ["primary", "size", "backgroundColor", "label", "icon"]);
    var mode = primary ? 'button--primary' : 'button--secondary';
    return (React.createElement("button", __assign({ type: "submit", className: ['button--container', "button--".concat(size), mode].join(' '), style: { backgroundColor: backgroundColor } }, props),
        label,
        icon));
};

var css_248z$6 = ".pool-card--container {\n  font-family: \"inter\", sans-serif;\n  background-color: #fff;\n  border-radius: 8px;\n  padding: 24px 20px;\n  width: 100%;\n  display: flex;\n  justify-content: space-between;\n}\n.pool-card--container .item1 {\n  grid-area: icon;\n}\n.pool-card--container .item1 img {\n  height: 40px;\n  width: 40px;\n}\n.pool-card--container .item2 {\n  grid-area: title;\n  font-weight: bold;\n  font-size: 14px;\n  color: #003c6e;\n}\n.pool-card--container .item3 {\n  grid-area: desc;\n  font-weight: normal;\n  font-size: small;\n  color: #003c6e;\n}\n.pool-card--container .pool--title {\n  display: grid;\n  grid-template-areas: \"icon title\" \"icon desc\";\n  gap: 0px 20px;\n}\n.pool-card--container .pool--stats {\n  display: flex;\n  flex-direction: column;\n  gap: 5px;\n}\n.pool-card--container .pool--stats svg {\n  height: 15px;\n  width: 15px;\n}\n.pool-card--container .pool--stats :nth-child(1) {\n  font-weight: bold;\n  font-size: 14px;\n  color: #003c6e;\n}\n.pool-card--container .pool--stats :nth-child(2) {\n  display: flex;\n  gap: 2px;\n  font-weight: normal;\n  font-size: small;\n  color: #003c6e;\n}\n.pool-card--container .pool--button {\n  display: flex;\n  align-items: center;\n  gap: 10px;\n}\n.pool-card--container .pool--button .button--secondary,\n.pool-card--container .pool--button .button--primary {\n  width: 130px;\n}";
styleInject(css_248z$6);

function getDefaultExportFromCjs (x) {
	return x && x.__esModule && Object.prototype.hasOwnProperty.call(x, 'default') ? x['default'] : x;
}

var propTypes = {exports: {}};

var reactIs = {exports: {}};

var reactIs_production_min = {};

/** @license React v16.13.1
 * react-is.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

var hasRequiredReactIs_production_min;

function requireReactIs_production_min () {
	if (hasRequiredReactIs_production_min) return reactIs_production_min;
	hasRequiredReactIs_production_min = 1;
var b="function"===typeof Symbol&&Symbol.for,c=b?Symbol.for("react.element"):60103,d=b?Symbol.for("react.portal"):60106,e=b?Symbol.for("react.fragment"):60107,f=b?Symbol.for("react.strict_mode"):60108,g=b?Symbol.for("react.profiler"):60114,h=b?Symbol.for("react.provider"):60109,k=b?Symbol.for("react.context"):60110,l=b?Symbol.for("react.async_mode"):60111,m=b?Symbol.for("react.concurrent_mode"):60111,n=b?Symbol.for("react.forward_ref"):60112,p=b?Symbol.for("react.suspense"):60113,q=b?
	Symbol.for("react.suspense_list"):60120,r=b?Symbol.for("react.memo"):60115,t=b?Symbol.for("react.lazy"):60116,v=b?Symbol.for("react.block"):60121,w=b?Symbol.for("react.fundamental"):60117,x=b?Symbol.for("react.responder"):60118,y=b?Symbol.for("react.scope"):60119;
	function z(a){if("object"===typeof a&&null!==a){var u=a.$$typeof;switch(u){case c:switch(a=a.type,a){case l:case m:case e:case g:case f:case p:return a;default:switch(a=a&&a.$$typeof,a){case k:case n:case t:case r:case h:return a;default:return u}}case d:return u}}}function A(a){return z(a)===m}reactIs_production_min.AsyncMode=l;reactIs_production_min.ConcurrentMode=m;reactIs_production_min.ContextConsumer=k;reactIs_production_min.ContextProvider=h;reactIs_production_min.Element=c;reactIs_production_min.ForwardRef=n;reactIs_production_min.Fragment=e;reactIs_production_min.Lazy=t;reactIs_production_min.Memo=r;reactIs_production_min.Portal=d;
	reactIs_production_min.Profiler=g;reactIs_production_min.StrictMode=f;reactIs_production_min.Suspense=p;reactIs_production_min.isAsyncMode=function(a){return A(a)||z(a)===l};reactIs_production_min.isConcurrentMode=A;reactIs_production_min.isContextConsumer=function(a){return z(a)===k};reactIs_production_min.isContextProvider=function(a){return z(a)===h};reactIs_production_min.isElement=function(a){return "object"===typeof a&&null!==a&&a.$$typeof===c};reactIs_production_min.isForwardRef=function(a){return z(a)===n};reactIs_production_min.isFragment=function(a){return z(a)===e};reactIs_production_min.isLazy=function(a){return z(a)===t};
	reactIs_production_min.isMemo=function(a){return z(a)===r};reactIs_production_min.isPortal=function(a){return z(a)===d};reactIs_production_min.isProfiler=function(a){return z(a)===g};reactIs_production_min.isStrictMode=function(a){return z(a)===f};reactIs_production_min.isSuspense=function(a){return z(a)===p};
	reactIs_production_min.isValidElementType=function(a){return "string"===typeof a||"function"===typeof a||a===e||a===m||a===g||a===f||a===p||a===q||"object"===typeof a&&null!==a&&(a.$$typeof===t||a.$$typeof===r||a.$$typeof===h||a.$$typeof===k||a.$$typeof===n||a.$$typeof===w||a.$$typeof===x||a.$$typeof===y||a.$$typeof===v)};reactIs_production_min.typeOf=z;
	return reactIs_production_min;
}

var reactIs_development = {};

/** @license React v16.13.1
 * react-is.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

var hasRequiredReactIs_development;

function requireReactIs_development () {
	if (hasRequiredReactIs_development) return reactIs_development;
	hasRequiredReactIs_development = 1;



	if (process.env.NODE_ENV !== "production") {
	  (function() {

	// The Symbol used to tag the ReactElement-like types. If there is no native Symbol
	// nor polyfill, then a plain number is used for performance.
	var hasSymbol = typeof Symbol === 'function' && Symbol.for;
	var REACT_ELEMENT_TYPE = hasSymbol ? Symbol.for('react.element') : 0xeac7;
	var REACT_PORTAL_TYPE = hasSymbol ? Symbol.for('react.portal') : 0xeaca;
	var REACT_FRAGMENT_TYPE = hasSymbol ? Symbol.for('react.fragment') : 0xeacb;
	var REACT_STRICT_MODE_TYPE = hasSymbol ? Symbol.for('react.strict_mode') : 0xeacc;
	var REACT_PROFILER_TYPE = hasSymbol ? Symbol.for('react.profiler') : 0xead2;
	var REACT_PROVIDER_TYPE = hasSymbol ? Symbol.for('react.provider') : 0xeacd;
	var REACT_CONTEXT_TYPE = hasSymbol ? Symbol.for('react.context') : 0xeace; // TODO: We don't use AsyncMode or ConcurrentMode anymore. They were temporary
	// (unstable) APIs that have been removed. Can we remove the symbols?

	var REACT_ASYNC_MODE_TYPE = hasSymbol ? Symbol.for('react.async_mode') : 0xeacf;
	var REACT_CONCURRENT_MODE_TYPE = hasSymbol ? Symbol.for('react.concurrent_mode') : 0xeacf;
	var REACT_FORWARD_REF_TYPE = hasSymbol ? Symbol.for('react.forward_ref') : 0xead0;
	var REACT_SUSPENSE_TYPE = hasSymbol ? Symbol.for('react.suspense') : 0xead1;
	var REACT_SUSPENSE_LIST_TYPE = hasSymbol ? Symbol.for('react.suspense_list') : 0xead8;
	var REACT_MEMO_TYPE = hasSymbol ? Symbol.for('react.memo') : 0xead3;
	var REACT_LAZY_TYPE = hasSymbol ? Symbol.for('react.lazy') : 0xead4;
	var REACT_BLOCK_TYPE = hasSymbol ? Symbol.for('react.block') : 0xead9;
	var REACT_FUNDAMENTAL_TYPE = hasSymbol ? Symbol.for('react.fundamental') : 0xead5;
	var REACT_RESPONDER_TYPE = hasSymbol ? Symbol.for('react.responder') : 0xead6;
	var REACT_SCOPE_TYPE = hasSymbol ? Symbol.for('react.scope') : 0xead7;

	function isValidElementType(type) {
	  return typeof type === 'string' || typeof type === 'function' || // Note: its typeof might be other than 'symbol' or 'number' if it's a polyfill.
	  type === REACT_FRAGMENT_TYPE || type === REACT_CONCURRENT_MODE_TYPE || type === REACT_PROFILER_TYPE || type === REACT_STRICT_MODE_TYPE || type === REACT_SUSPENSE_TYPE || type === REACT_SUSPENSE_LIST_TYPE || typeof type === 'object' && type !== null && (type.$$typeof === REACT_LAZY_TYPE || type.$$typeof === REACT_MEMO_TYPE || type.$$typeof === REACT_PROVIDER_TYPE || type.$$typeof === REACT_CONTEXT_TYPE || type.$$typeof === REACT_FORWARD_REF_TYPE || type.$$typeof === REACT_FUNDAMENTAL_TYPE || type.$$typeof === REACT_RESPONDER_TYPE || type.$$typeof === REACT_SCOPE_TYPE || type.$$typeof === REACT_BLOCK_TYPE);
	}

	function typeOf(object) {
	  if (typeof object === 'object' && object !== null) {
	    var $$typeof = object.$$typeof;

	    switch ($$typeof) {
	      case REACT_ELEMENT_TYPE:
	        var type = object.type;

	        switch (type) {
	          case REACT_ASYNC_MODE_TYPE:
	          case REACT_CONCURRENT_MODE_TYPE:
	          case REACT_FRAGMENT_TYPE:
	          case REACT_PROFILER_TYPE:
	          case REACT_STRICT_MODE_TYPE:
	          case REACT_SUSPENSE_TYPE:
	            return type;

	          default:
	            var $$typeofType = type && type.$$typeof;

	            switch ($$typeofType) {
	              case REACT_CONTEXT_TYPE:
	              case REACT_FORWARD_REF_TYPE:
	              case REACT_LAZY_TYPE:
	              case REACT_MEMO_TYPE:
	              case REACT_PROVIDER_TYPE:
	                return $$typeofType;

	              default:
	                return $$typeof;
	            }

	        }

	      case REACT_PORTAL_TYPE:
	        return $$typeof;
	    }
	  }

	  return undefined;
	} // AsyncMode is deprecated along with isAsyncMode

	var AsyncMode = REACT_ASYNC_MODE_TYPE;
	var ConcurrentMode = REACT_CONCURRENT_MODE_TYPE;
	var ContextConsumer = REACT_CONTEXT_TYPE;
	var ContextProvider = REACT_PROVIDER_TYPE;
	var Element = REACT_ELEMENT_TYPE;
	var ForwardRef = REACT_FORWARD_REF_TYPE;
	var Fragment = REACT_FRAGMENT_TYPE;
	var Lazy = REACT_LAZY_TYPE;
	var Memo = REACT_MEMO_TYPE;
	var Portal = REACT_PORTAL_TYPE;
	var Profiler = REACT_PROFILER_TYPE;
	var StrictMode = REACT_STRICT_MODE_TYPE;
	var Suspense = REACT_SUSPENSE_TYPE;
	var hasWarnedAboutDeprecatedIsAsyncMode = false; // AsyncMode should be deprecated

	function isAsyncMode(object) {
	  {
	    if (!hasWarnedAboutDeprecatedIsAsyncMode) {
	      hasWarnedAboutDeprecatedIsAsyncMode = true; // Using console['warn'] to evade Babel and ESLint

	      console['warn']('The ReactIs.isAsyncMode() alias has been deprecated, ' + 'and will be removed in React 17+. Update your code to use ' + 'ReactIs.isConcurrentMode() instead. It has the exact same API.');
	    }
	  }

	  return isConcurrentMode(object) || typeOf(object) === REACT_ASYNC_MODE_TYPE;
	}
	function isConcurrentMode(object) {
	  return typeOf(object) === REACT_CONCURRENT_MODE_TYPE;
	}
	function isContextConsumer(object) {
	  return typeOf(object) === REACT_CONTEXT_TYPE;
	}
	function isContextProvider(object) {
	  return typeOf(object) === REACT_PROVIDER_TYPE;
	}
	function isElement(object) {
	  return typeof object === 'object' && object !== null && object.$$typeof === REACT_ELEMENT_TYPE;
	}
	function isForwardRef(object) {
	  return typeOf(object) === REACT_FORWARD_REF_TYPE;
	}
	function isFragment(object) {
	  return typeOf(object) === REACT_FRAGMENT_TYPE;
	}
	function isLazy(object) {
	  return typeOf(object) === REACT_LAZY_TYPE;
	}
	function isMemo(object) {
	  return typeOf(object) === REACT_MEMO_TYPE;
	}
	function isPortal(object) {
	  return typeOf(object) === REACT_PORTAL_TYPE;
	}
	function isProfiler(object) {
	  return typeOf(object) === REACT_PROFILER_TYPE;
	}
	function isStrictMode(object) {
	  return typeOf(object) === REACT_STRICT_MODE_TYPE;
	}
	function isSuspense(object) {
	  return typeOf(object) === REACT_SUSPENSE_TYPE;
	}

	reactIs_development.AsyncMode = AsyncMode;
	reactIs_development.ConcurrentMode = ConcurrentMode;
	reactIs_development.ContextConsumer = ContextConsumer;
	reactIs_development.ContextProvider = ContextProvider;
	reactIs_development.Element = Element;
	reactIs_development.ForwardRef = ForwardRef;
	reactIs_development.Fragment = Fragment;
	reactIs_development.Lazy = Lazy;
	reactIs_development.Memo = Memo;
	reactIs_development.Portal = Portal;
	reactIs_development.Profiler = Profiler;
	reactIs_development.StrictMode = StrictMode;
	reactIs_development.Suspense = Suspense;
	reactIs_development.isAsyncMode = isAsyncMode;
	reactIs_development.isConcurrentMode = isConcurrentMode;
	reactIs_development.isContextConsumer = isContextConsumer;
	reactIs_development.isContextProvider = isContextProvider;
	reactIs_development.isElement = isElement;
	reactIs_development.isForwardRef = isForwardRef;
	reactIs_development.isFragment = isFragment;
	reactIs_development.isLazy = isLazy;
	reactIs_development.isMemo = isMemo;
	reactIs_development.isPortal = isPortal;
	reactIs_development.isProfiler = isProfiler;
	reactIs_development.isStrictMode = isStrictMode;
	reactIs_development.isSuspense = isSuspense;
	reactIs_development.isValidElementType = isValidElementType;
	reactIs_development.typeOf = typeOf;
	  })();
	}
	return reactIs_development;
}

var hasRequiredReactIs;

function requireReactIs () {
	if (hasRequiredReactIs) return reactIs.exports;
	hasRequiredReactIs = 1;

	if (process.env.NODE_ENV === 'production') {
	  reactIs.exports = requireReactIs_production_min();
	} else {
	  reactIs.exports = requireReactIs_development();
	}
	return reactIs.exports;
}

/*
object-assign
(c) Sindre Sorhus
@license MIT
*/

var objectAssign;
var hasRequiredObjectAssign;

function requireObjectAssign () {
	if (hasRequiredObjectAssign) return objectAssign;
	hasRequiredObjectAssign = 1;
	/* eslint-disable no-unused-vars */
	var getOwnPropertySymbols = Object.getOwnPropertySymbols;
	var hasOwnProperty = Object.prototype.hasOwnProperty;
	var propIsEnumerable = Object.prototype.propertyIsEnumerable;

	function toObject(val) {
		if (val === null || val === undefined) {
			throw new TypeError('Object.assign cannot be called with null or undefined');
		}

		return Object(val);
	}

	function shouldUseNative() {
		try {
			if (!Object.assign) {
				return false;
			}

			// Detect buggy property enumeration order in older V8 versions.

			// https://bugs.chromium.org/p/v8/issues/detail?id=4118
			var test1 = new String('abc');  // eslint-disable-line no-new-wrappers
			test1[5] = 'de';
			if (Object.getOwnPropertyNames(test1)[0] === '5') {
				return false;
			}

			// https://bugs.chromium.org/p/v8/issues/detail?id=3056
			var test2 = {};
			for (var i = 0; i < 10; i++) {
				test2['_' + String.fromCharCode(i)] = i;
			}
			var order2 = Object.getOwnPropertyNames(test2).map(function (n) {
				return test2[n];
			});
			if (order2.join('') !== '0123456789') {
				return false;
			}

			// https://bugs.chromium.org/p/v8/issues/detail?id=3056
			var test3 = {};
			'abcdefghijklmnopqrst'.split('').forEach(function (letter) {
				test3[letter] = letter;
			});
			if (Object.keys(Object.assign({}, test3)).join('') !==
					'abcdefghijklmnopqrst') {
				return false;
			}

			return true;
		} catch (err) {
			// We don't expect any of the above to throw, but better to be safe.
			return false;
		}
	}

	objectAssign = shouldUseNative() ? Object.assign : function (target, source) {
		var from;
		var to = toObject(target);
		var symbols;

		for (var s = 1; s < arguments.length; s++) {
			from = Object(arguments[s]);

			for (var key in from) {
				if (hasOwnProperty.call(from, key)) {
					to[key] = from[key];
				}
			}

			if (getOwnPropertySymbols) {
				symbols = getOwnPropertySymbols(from);
				for (var i = 0; i < symbols.length; i++) {
					if (propIsEnumerable.call(from, symbols[i])) {
						to[symbols[i]] = from[symbols[i]];
					}
				}
			}
		}

		return to;
	};
	return objectAssign;
}

/**
 * Copyright (c) 2013-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

var ReactPropTypesSecret_1;
var hasRequiredReactPropTypesSecret;

function requireReactPropTypesSecret () {
	if (hasRequiredReactPropTypesSecret) return ReactPropTypesSecret_1;
	hasRequiredReactPropTypesSecret = 1;

	var ReactPropTypesSecret = 'SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED';

	ReactPropTypesSecret_1 = ReactPropTypesSecret;
	return ReactPropTypesSecret_1;
}

var has;
var hasRequiredHas;

function requireHas () {
	if (hasRequiredHas) return has;
	hasRequiredHas = 1;
	has = Function.call.bind(Object.prototype.hasOwnProperty);
	return has;
}

/**
 * Copyright (c) 2013-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

var checkPropTypes_1;
var hasRequiredCheckPropTypes;

function requireCheckPropTypes () {
	if (hasRequiredCheckPropTypes) return checkPropTypes_1;
	hasRequiredCheckPropTypes = 1;

	var printWarning = function() {};

	if (process.env.NODE_ENV !== 'production') {
	  var ReactPropTypesSecret = requireReactPropTypesSecret();
	  var loggedTypeFailures = {};
	  var has = requireHas();

	  printWarning = function(text) {
	    var message = 'Warning: ' + text;
	    if (typeof console !== 'undefined') {
	      console.error(message);
	    }
	    try {
	      // --- Welcome to debugging React ---
	      // This error was thrown as a convenience so that you can use this stack
	      // to find the callsite that caused this warning to fire.
	      throw new Error(message);
	    } catch (x) { /**/ }
	  };
	}

	/**
	 * Assert that the values match with the type specs.
	 * Error messages are memorized and will only be shown once.
	 *
	 * @param {object} typeSpecs Map of name to a ReactPropType
	 * @param {object} values Runtime values that need to be type-checked
	 * @param {string} location e.g. "prop", "context", "child context"
	 * @param {string} componentName Name of the component for error messages.
	 * @param {?Function} getStack Returns the component stack.
	 * @private
	 */
	function checkPropTypes(typeSpecs, values, location, componentName, getStack) {
	  if (process.env.NODE_ENV !== 'production') {
	    for (var typeSpecName in typeSpecs) {
	      if (has(typeSpecs, typeSpecName)) {
	        var error;
	        // Prop type validation may throw. In case they do, we don't want to
	        // fail the render phase where it didn't fail before. So we log it.
	        // After these have been cleaned up, we'll let them throw.
	        try {
	          // This is intentionally an invariant that gets caught. It's the same
	          // behavior as without this statement except with a better message.
	          if (typeof typeSpecs[typeSpecName] !== 'function') {
	            var err = Error(
	              (componentName || 'React class') + ': ' + location + ' type `' + typeSpecName + '` is invalid; ' +
	              'it must be a function, usually from the `prop-types` package, but received `' + typeof typeSpecs[typeSpecName] + '`.' +
	              'This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.'
	            );
	            err.name = 'Invariant Violation';
	            throw err;
	          }
	          error = typeSpecs[typeSpecName](values, typeSpecName, componentName, location, null, ReactPropTypesSecret);
	        } catch (ex) {
	          error = ex;
	        }
	        if (error && !(error instanceof Error)) {
	          printWarning(
	            (componentName || 'React class') + ': type specification of ' +
	            location + ' `' + typeSpecName + '` is invalid; the type checker ' +
	            'function must return `null` or an `Error` but returned a ' + typeof error + '. ' +
	            'You may have forgotten to pass an argument to the type checker ' +
	            'creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and ' +
	            'shape all require an argument).'
	          );
	        }
	        if (error instanceof Error && !(error.message in loggedTypeFailures)) {
	          // Only monitor this failure once because there tends to be a lot of the
	          // same error.
	          loggedTypeFailures[error.message] = true;

	          var stack = getStack ? getStack() : '';

	          printWarning(
	            'Failed ' + location + ' type: ' + error.message + (stack != null ? stack : '')
	          );
	        }
	      }
	    }
	  }
	}

	/**
	 * Resets warning cache when testing.
	 *
	 * @private
	 */
	checkPropTypes.resetWarningCache = function() {
	  if (process.env.NODE_ENV !== 'production') {
	    loggedTypeFailures = {};
	  }
	};

	checkPropTypes_1 = checkPropTypes;
	return checkPropTypes_1;
}

/**
 * Copyright (c) 2013-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

var factoryWithTypeCheckers;
var hasRequiredFactoryWithTypeCheckers;

function requireFactoryWithTypeCheckers () {
	if (hasRequiredFactoryWithTypeCheckers) return factoryWithTypeCheckers;
	hasRequiredFactoryWithTypeCheckers = 1;

	var ReactIs = requireReactIs();
	var assign = requireObjectAssign();

	var ReactPropTypesSecret = requireReactPropTypesSecret();
	var has = requireHas();
	var checkPropTypes = requireCheckPropTypes();

	var printWarning = function() {};

	if (process.env.NODE_ENV !== 'production') {
	  printWarning = function(text) {
	    var message = 'Warning: ' + text;
	    if (typeof console !== 'undefined') {
	      console.error(message);
	    }
	    try {
	      // --- Welcome to debugging React ---
	      // This error was thrown as a convenience so that you can use this stack
	      // to find the callsite that caused this warning to fire.
	      throw new Error(message);
	    } catch (x) {}
	  };
	}

	function emptyFunctionThatReturnsNull() {
	  return null;
	}

	factoryWithTypeCheckers = function(isValidElement, throwOnDirectAccess) {
	  /* global Symbol */
	  var ITERATOR_SYMBOL = typeof Symbol === 'function' && Symbol.iterator;
	  var FAUX_ITERATOR_SYMBOL = '@@iterator'; // Before Symbol spec.

	  /**
	   * Returns the iterator method function contained on the iterable object.
	   *
	   * Be sure to invoke the function with the iterable as context:
	   *
	   *     var iteratorFn = getIteratorFn(myIterable);
	   *     if (iteratorFn) {
	   *       var iterator = iteratorFn.call(myIterable);
	   *       ...
	   *     }
	   *
	   * @param {?object} maybeIterable
	   * @return {?function}
	   */
	  function getIteratorFn(maybeIterable) {
	    var iteratorFn = maybeIterable && (ITERATOR_SYMBOL && maybeIterable[ITERATOR_SYMBOL] || maybeIterable[FAUX_ITERATOR_SYMBOL]);
	    if (typeof iteratorFn === 'function') {
	      return iteratorFn;
	    }
	  }

	  /**
	   * Collection of methods that allow declaration and validation of props that are
	   * supplied to React components. Example usage:
	   *
	   *   var Props = require('ReactPropTypes');
	   *   var MyArticle = React.createClass({
	   *     propTypes: {
	   *       // An optional string prop named "description".
	   *       description: Props.string,
	   *
	   *       // A required enum prop named "category".
	   *       category: Props.oneOf(['News','Photos']).isRequired,
	   *
	   *       // A prop named "dialog" that requires an instance of Dialog.
	   *       dialog: Props.instanceOf(Dialog).isRequired
	   *     },
	   *     render: function() { ... }
	   *   });
	   *
	   * A more formal specification of how these methods are used:
	   *
	   *   type := array|bool|func|object|number|string|oneOf([...])|instanceOf(...)
	   *   decl := ReactPropTypes.{type}(.isRequired)?
	   *
	   * Each and every declaration produces a function with the same signature. This
	   * allows the creation of custom validation functions. For example:
	   *
	   *  var MyLink = React.createClass({
	   *    propTypes: {
	   *      // An optional string or URI prop named "href".
	   *      href: function(props, propName, componentName) {
	   *        var propValue = props[propName];
	   *        if (propValue != null && typeof propValue !== 'string' &&
	   *            !(propValue instanceof URI)) {
	   *          return new Error(
	   *            'Expected a string or an URI for ' + propName + ' in ' +
	   *            componentName
	   *          );
	   *        }
	   *      }
	   *    },
	   *    render: function() {...}
	   *  });
	   *
	   * @internal
	   */

	  var ANONYMOUS = '<<anonymous>>';

	  // Important!
	  // Keep this list in sync with production version in `./factoryWithThrowingShims.js`.
	  var ReactPropTypes = {
	    array: createPrimitiveTypeChecker('array'),
	    bigint: createPrimitiveTypeChecker('bigint'),
	    bool: createPrimitiveTypeChecker('boolean'),
	    func: createPrimitiveTypeChecker('function'),
	    number: createPrimitiveTypeChecker('number'),
	    object: createPrimitiveTypeChecker('object'),
	    string: createPrimitiveTypeChecker('string'),
	    symbol: createPrimitiveTypeChecker('symbol'),

	    any: createAnyTypeChecker(),
	    arrayOf: createArrayOfTypeChecker,
	    element: createElementTypeChecker(),
	    elementType: createElementTypeTypeChecker(),
	    instanceOf: createInstanceTypeChecker,
	    node: createNodeChecker(),
	    objectOf: createObjectOfTypeChecker,
	    oneOf: createEnumTypeChecker,
	    oneOfType: createUnionTypeChecker,
	    shape: createShapeTypeChecker,
	    exact: createStrictShapeTypeChecker,
	  };

	  /**
	   * inlined Object.is polyfill to avoid requiring consumers ship their own
	   * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is
	   */
	  /*eslint-disable no-self-compare*/
	  function is(x, y) {
	    // SameValue algorithm
	    if (x === y) {
	      // Steps 1-5, 7-10
	      // Steps 6.b-6.e: +0 != -0
	      return x !== 0 || 1 / x === 1 / y;
	    } else {
	      // Step 6.a: NaN == NaN
	      return x !== x && y !== y;
	    }
	  }
	  /*eslint-enable no-self-compare*/

	  /**
	   * We use an Error-like object for backward compatibility as people may call
	   * PropTypes directly and inspect their output. However, we don't use real
	   * Errors anymore. We don't inspect their stack anyway, and creating them
	   * is prohibitively expensive if they are created too often, such as what
	   * happens in oneOfType() for any type before the one that matched.
	   */
	  function PropTypeError(message, data) {
	    this.message = message;
	    this.data = data && typeof data === 'object' ? data: {};
	    this.stack = '';
	  }
	  // Make `instanceof Error` still work for returned errors.
	  PropTypeError.prototype = Error.prototype;

	  function createChainableTypeChecker(validate) {
	    if (process.env.NODE_ENV !== 'production') {
	      var manualPropTypeCallCache = {};
	      var manualPropTypeWarningCount = 0;
	    }
	    function checkType(isRequired, props, propName, componentName, location, propFullName, secret) {
	      componentName = componentName || ANONYMOUS;
	      propFullName = propFullName || propName;

	      if (secret !== ReactPropTypesSecret) {
	        if (throwOnDirectAccess) {
	          // New behavior only for users of `prop-types` package
	          var err = new Error(
	            'Calling PropTypes validators directly is not supported by the `prop-types` package. ' +
	            'Use `PropTypes.checkPropTypes()` to call them. ' +
	            'Read more at http://fb.me/use-check-prop-types'
	          );
	          err.name = 'Invariant Violation';
	          throw err;
	        } else if (process.env.NODE_ENV !== 'production' && typeof console !== 'undefined') {
	          // Old behavior for people using React.PropTypes
	          var cacheKey = componentName + ':' + propName;
	          if (
	            !manualPropTypeCallCache[cacheKey] &&
	            // Avoid spamming the console because they are often not actionable except for lib authors
	            manualPropTypeWarningCount < 3
	          ) {
	            printWarning(
	              'You are manually calling a React.PropTypes validation ' +
	              'function for the `' + propFullName + '` prop on `' + componentName + '`. This is deprecated ' +
	              'and will throw in the standalone `prop-types` package. ' +
	              'You may be seeing this warning due to a third-party PropTypes ' +
	              'library. See https://fb.me/react-warning-dont-call-proptypes ' + 'for details.'
	            );
	            manualPropTypeCallCache[cacheKey] = true;
	            manualPropTypeWarningCount++;
	          }
	        }
	      }
	      if (props[propName] == null) {
	        if (isRequired) {
	          if (props[propName] === null) {
	            return new PropTypeError('The ' + location + ' `' + propFullName + '` is marked as required ' + ('in `' + componentName + '`, but its value is `null`.'));
	          }
	          return new PropTypeError('The ' + location + ' `' + propFullName + '` is marked as required in ' + ('`' + componentName + '`, but its value is `undefined`.'));
	        }
	        return null;
	      } else {
	        return validate(props, propName, componentName, location, propFullName);
	      }
	    }

	    var chainedCheckType = checkType.bind(null, false);
	    chainedCheckType.isRequired = checkType.bind(null, true);

	    return chainedCheckType;
	  }

	  function createPrimitiveTypeChecker(expectedType) {
	    function validate(props, propName, componentName, location, propFullName, secret) {
	      var propValue = props[propName];
	      var propType = getPropType(propValue);
	      if (propType !== expectedType) {
	        // `propValue` being instance of, say, date/regexp, pass the 'object'
	        // check, but we can offer a more precise error message here rather than
	        // 'of type `object`'.
	        var preciseType = getPreciseType(propValue);

	        return new PropTypeError(
	          'Invalid ' + location + ' `' + propFullName + '` of type ' + ('`' + preciseType + '` supplied to `' + componentName + '`, expected ') + ('`' + expectedType + '`.'),
	          {expectedType: expectedType}
	        );
	      }
	      return null;
	    }
	    return createChainableTypeChecker(validate);
	  }

	  function createAnyTypeChecker() {
	    return createChainableTypeChecker(emptyFunctionThatReturnsNull);
	  }

	  function createArrayOfTypeChecker(typeChecker) {
	    function validate(props, propName, componentName, location, propFullName) {
	      if (typeof typeChecker !== 'function') {
	        return new PropTypeError('Property `' + propFullName + '` of component `' + componentName + '` has invalid PropType notation inside arrayOf.');
	      }
	      var propValue = props[propName];
	      if (!Array.isArray(propValue)) {
	        var propType = getPropType(propValue);
	        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type ' + ('`' + propType + '` supplied to `' + componentName + '`, expected an array.'));
	      }
	      for (var i = 0; i < propValue.length; i++) {
	        var error = typeChecker(propValue, i, componentName, location, propFullName + '[' + i + ']', ReactPropTypesSecret);
	        if (error instanceof Error) {
	          return error;
	        }
	      }
	      return null;
	    }
	    return createChainableTypeChecker(validate);
	  }

	  function createElementTypeChecker() {
	    function validate(props, propName, componentName, location, propFullName) {
	      var propValue = props[propName];
	      if (!isValidElement(propValue)) {
	        var propType = getPropType(propValue);
	        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type ' + ('`' + propType + '` supplied to `' + componentName + '`, expected a single ReactElement.'));
	      }
	      return null;
	    }
	    return createChainableTypeChecker(validate);
	  }

	  function createElementTypeTypeChecker() {
	    function validate(props, propName, componentName, location, propFullName) {
	      var propValue = props[propName];
	      if (!ReactIs.isValidElementType(propValue)) {
	        var propType = getPropType(propValue);
	        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type ' + ('`' + propType + '` supplied to `' + componentName + '`, expected a single ReactElement type.'));
	      }
	      return null;
	    }
	    return createChainableTypeChecker(validate);
	  }

	  function createInstanceTypeChecker(expectedClass) {
	    function validate(props, propName, componentName, location, propFullName) {
	      if (!(props[propName] instanceof expectedClass)) {
	        var expectedClassName = expectedClass.name || ANONYMOUS;
	        var actualClassName = getClassName(props[propName]);
	        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type ' + ('`' + actualClassName + '` supplied to `' + componentName + '`, expected ') + ('instance of `' + expectedClassName + '`.'));
	      }
	      return null;
	    }
	    return createChainableTypeChecker(validate);
	  }

	  function createEnumTypeChecker(expectedValues) {
	    if (!Array.isArray(expectedValues)) {
	      if (process.env.NODE_ENV !== 'production') {
	        if (arguments.length > 1) {
	          printWarning(
	            'Invalid arguments supplied to oneOf, expected an array, got ' + arguments.length + ' arguments. ' +
	            'A common mistake is to write oneOf(x, y, z) instead of oneOf([x, y, z]).'
	          );
	        } else {
	          printWarning('Invalid argument supplied to oneOf, expected an array.');
	        }
	      }
	      return emptyFunctionThatReturnsNull;
	    }

	    function validate(props, propName, componentName, location, propFullName) {
	      var propValue = props[propName];
	      for (var i = 0; i < expectedValues.length; i++) {
	        if (is(propValue, expectedValues[i])) {
	          return null;
	        }
	      }

	      var valuesString = JSON.stringify(expectedValues, function replacer(key, value) {
	        var type = getPreciseType(value);
	        if (type === 'symbol') {
	          return String(value);
	        }
	        return value;
	      });
	      return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of value `' + String(propValue) + '` ' + ('supplied to `' + componentName + '`, expected one of ' + valuesString + '.'));
	    }
	    return createChainableTypeChecker(validate);
	  }

	  function createObjectOfTypeChecker(typeChecker) {
	    function validate(props, propName, componentName, location, propFullName) {
	      if (typeof typeChecker !== 'function') {
	        return new PropTypeError('Property `' + propFullName + '` of component `' + componentName + '` has invalid PropType notation inside objectOf.');
	      }
	      var propValue = props[propName];
	      var propType = getPropType(propValue);
	      if (propType !== 'object') {
	        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type ' + ('`' + propType + '` supplied to `' + componentName + '`, expected an object.'));
	      }
	      for (var key in propValue) {
	        if (has(propValue, key)) {
	          var error = typeChecker(propValue, key, componentName, location, propFullName + '.' + key, ReactPropTypesSecret);
	          if (error instanceof Error) {
	            return error;
	          }
	        }
	      }
	      return null;
	    }
	    return createChainableTypeChecker(validate);
	  }

	  function createUnionTypeChecker(arrayOfTypeCheckers) {
	    if (!Array.isArray(arrayOfTypeCheckers)) {
	      process.env.NODE_ENV !== 'production' ? printWarning('Invalid argument supplied to oneOfType, expected an instance of array.') : void 0;
	      return emptyFunctionThatReturnsNull;
	    }

	    for (var i = 0; i < arrayOfTypeCheckers.length; i++) {
	      var checker = arrayOfTypeCheckers[i];
	      if (typeof checker !== 'function') {
	        printWarning(
	          'Invalid argument supplied to oneOfType. Expected an array of check functions, but ' +
	          'received ' + getPostfixForTypeWarning(checker) + ' at index ' + i + '.'
	        );
	        return emptyFunctionThatReturnsNull;
	      }
	    }

	    function validate(props, propName, componentName, location, propFullName) {
	      var expectedTypes = [];
	      for (var i = 0; i < arrayOfTypeCheckers.length; i++) {
	        var checker = arrayOfTypeCheckers[i];
	        var checkerResult = checker(props, propName, componentName, location, propFullName, ReactPropTypesSecret);
	        if (checkerResult == null) {
	          return null;
	        }
	        if (checkerResult.data && has(checkerResult.data, 'expectedType')) {
	          expectedTypes.push(checkerResult.data.expectedType);
	        }
	      }
	      var expectedTypesMessage = (expectedTypes.length > 0) ? ', expected one of type [' + expectedTypes.join(', ') + ']': '';
	      return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` supplied to ' + ('`' + componentName + '`' + expectedTypesMessage + '.'));
	    }
	    return createChainableTypeChecker(validate);
	  }

	  function createNodeChecker() {
	    function validate(props, propName, componentName, location, propFullName) {
	      if (!isNode(props[propName])) {
	        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` supplied to ' + ('`' + componentName + '`, expected a ReactNode.'));
	      }
	      return null;
	    }
	    return createChainableTypeChecker(validate);
	  }

	  function invalidValidatorError(componentName, location, propFullName, key, type) {
	    return new PropTypeError(
	      (componentName || 'React class') + ': ' + location + ' type `' + propFullName + '.' + key + '` is invalid; ' +
	      'it must be a function, usually from the `prop-types` package, but received `' + type + '`.'
	    );
	  }

	  function createShapeTypeChecker(shapeTypes) {
	    function validate(props, propName, componentName, location, propFullName) {
	      var propValue = props[propName];
	      var propType = getPropType(propValue);
	      if (propType !== 'object') {
	        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type `' + propType + '` ' + ('supplied to `' + componentName + '`, expected `object`.'));
	      }
	      for (var key in shapeTypes) {
	        var checker = shapeTypes[key];
	        if (typeof checker !== 'function') {
	          return invalidValidatorError(componentName, location, propFullName, key, getPreciseType(checker));
	        }
	        var error = checker(propValue, key, componentName, location, propFullName + '.' + key, ReactPropTypesSecret);
	        if (error) {
	          return error;
	        }
	      }
	      return null;
	    }
	    return createChainableTypeChecker(validate);
	  }

	  function createStrictShapeTypeChecker(shapeTypes) {
	    function validate(props, propName, componentName, location, propFullName) {
	      var propValue = props[propName];
	      var propType = getPropType(propValue);
	      if (propType !== 'object') {
	        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type `' + propType + '` ' + ('supplied to `' + componentName + '`, expected `object`.'));
	      }
	      // We need to check all keys in case some are required but missing from props.
	      var allKeys = assign({}, props[propName], shapeTypes);
	      for (var key in allKeys) {
	        var checker = shapeTypes[key];
	        if (has(shapeTypes, key) && typeof checker !== 'function') {
	          return invalidValidatorError(componentName, location, propFullName, key, getPreciseType(checker));
	        }
	        if (!checker) {
	          return new PropTypeError(
	            'Invalid ' + location + ' `' + propFullName + '` key `' + key + '` supplied to `' + componentName + '`.' +
	            '\nBad object: ' + JSON.stringify(props[propName], null, '  ') +
	            '\nValid keys: ' + JSON.stringify(Object.keys(shapeTypes), null, '  ')
	          );
	        }
	        var error = checker(propValue, key, componentName, location, propFullName + '.' + key, ReactPropTypesSecret);
	        if (error) {
	          return error;
	        }
	      }
	      return null;
	    }

	    return createChainableTypeChecker(validate);
	  }

	  function isNode(propValue) {
	    switch (typeof propValue) {
	      case 'number':
	      case 'string':
	      case 'undefined':
	        return true;
	      case 'boolean':
	        return !propValue;
	      case 'object':
	        if (Array.isArray(propValue)) {
	          return propValue.every(isNode);
	        }
	        if (propValue === null || isValidElement(propValue)) {
	          return true;
	        }

	        var iteratorFn = getIteratorFn(propValue);
	        if (iteratorFn) {
	          var iterator = iteratorFn.call(propValue);
	          var step;
	          if (iteratorFn !== propValue.entries) {
	            while (!(step = iterator.next()).done) {
	              if (!isNode(step.value)) {
	                return false;
	              }
	            }
	          } else {
	            // Iterator will provide entry [k,v] tuples rather than values.
	            while (!(step = iterator.next()).done) {
	              var entry = step.value;
	              if (entry) {
	                if (!isNode(entry[1])) {
	                  return false;
	                }
	              }
	            }
	          }
	        } else {
	          return false;
	        }

	        return true;
	      default:
	        return false;
	    }
	  }

	  function isSymbol(propType, propValue) {
	    // Native Symbol.
	    if (propType === 'symbol') {
	      return true;
	    }

	    // falsy value can't be a Symbol
	    if (!propValue) {
	      return false;
	    }

	    // 19.4.3.5 Symbol.prototype[@@toStringTag] === 'Symbol'
	    if (propValue['@@toStringTag'] === 'Symbol') {
	      return true;
	    }

	    // Fallback for non-spec compliant Symbols which are polyfilled.
	    if (typeof Symbol === 'function' && propValue instanceof Symbol) {
	      return true;
	    }

	    return false;
	  }

	  // Equivalent of `typeof` but with special handling for array and regexp.
	  function getPropType(propValue) {
	    var propType = typeof propValue;
	    if (Array.isArray(propValue)) {
	      return 'array';
	    }
	    if (propValue instanceof RegExp) {
	      // Old webkits (at least until Android 4.0) return 'function' rather than
	      // 'object' for typeof a RegExp. We'll normalize this here so that /bla/
	      // passes PropTypes.object.
	      return 'object';
	    }
	    if (isSymbol(propType, propValue)) {
	      return 'symbol';
	    }
	    return propType;
	  }

	  // This handles more types than `getPropType`. Only used for error messages.
	  // See `createPrimitiveTypeChecker`.
	  function getPreciseType(propValue) {
	    if (typeof propValue === 'undefined' || propValue === null) {
	      return '' + propValue;
	    }
	    var propType = getPropType(propValue);
	    if (propType === 'object') {
	      if (propValue instanceof Date) {
	        return 'date';
	      } else if (propValue instanceof RegExp) {
	        return 'regexp';
	      }
	    }
	    return propType;
	  }

	  // Returns a string that is postfixed to a warning about an invalid type.
	  // For example, "undefined" or "of type array"
	  function getPostfixForTypeWarning(value) {
	    var type = getPreciseType(value);
	    switch (type) {
	      case 'array':
	      case 'object':
	        return 'an ' + type;
	      case 'boolean':
	      case 'date':
	      case 'regexp':
	        return 'a ' + type;
	      default:
	        return type;
	    }
	  }

	  // Returns class name of the object, if any.
	  function getClassName(propValue) {
	    if (!propValue.constructor || !propValue.constructor.name) {
	      return ANONYMOUS;
	    }
	    return propValue.constructor.name;
	  }

	  ReactPropTypes.checkPropTypes = checkPropTypes;
	  ReactPropTypes.resetWarningCache = checkPropTypes.resetWarningCache;
	  ReactPropTypes.PropTypes = ReactPropTypes;

	  return ReactPropTypes;
	};
	return factoryWithTypeCheckers;
}

/**
 * Copyright (c) 2013-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

var factoryWithThrowingShims;
var hasRequiredFactoryWithThrowingShims;

function requireFactoryWithThrowingShims () {
	if (hasRequiredFactoryWithThrowingShims) return factoryWithThrowingShims;
	hasRequiredFactoryWithThrowingShims = 1;

	var ReactPropTypesSecret = requireReactPropTypesSecret();

	function emptyFunction() {}
	function emptyFunctionWithReset() {}
	emptyFunctionWithReset.resetWarningCache = emptyFunction;

	factoryWithThrowingShims = function() {
	  function shim(props, propName, componentName, location, propFullName, secret) {
	    if (secret === ReactPropTypesSecret) {
	      // It is still safe when called from React.
	      return;
	    }
	    var err = new Error(
	      'Calling PropTypes validators directly is not supported by the `prop-types` package. ' +
	      'Use PropTypes.checkPropTypes() to call them. ' +
	      'Read more at http://fb.me/use-check-prop-types'
	    );
	    err.name = 'Invariant Violation';
	    throw err;
	  }	  shim.isRequired = shim;
	  function getShim() {
	    return shim;
	  }	  // Important!
	  // Keep this list in sync with production version in `./factoryWithTypeCheckers.js`.
	  var ReactPropTypes = {
	    array: shim,
	    bigint: shim,
	    bool: shim,
	    func: shim,
	    number: shim,
	    object: shim,
	    string: shim,
	    symbol: shim,

	    any: shim,
	    arrayOf: getShim,
	    element: shim,
	    elementType: shim,
	    instanceOf: getShim,
	    node: shim,
	    objectOf: getShim,
	    oneOf: getShim,
	    oneOfType: getShim,
	    shape: getShim,
	    exact: getShim,

	    checkPropTypes: emptyFunctionWithReset,
	    resetWarningCache: emptyFunction
	  };

	  ReactPropTypes.PropTypes = ReactPropTypes;

	  return ReactPropTypes;
	};
	return factoryWithThrowingShims;
}

/**
 * Copyright (c) 2013-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

if (process.env.NODE_ENV !== 'production') {
  var ReactIs = requireReactIs();

  // By explicitly using `prop-types` you are opting into new development behavior.
  // http://fb.me/prop-types-in-prod
  var throwOnDirectAccess = true;
  propTypes.exports = requireFactoryWithTypeCheckers()(ReactIs.isElement, throwOnDirectAccess);
} else {
  // By explicitly using `prop-types` you are opting into new production behavior.
  // http://fb.me/prop-types-in-prod
  propTypes.exports = requireFactoryWithThrowingShims()();
}

var propTypesExports = propTypes.exports;
var PropTypes = /*@__PURE__*/getDefaultExportFromCjs(propTypesExports);

function _extends$7() { _extends$7 = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$7.apply(this, arguments); }

function _objectWithoutProperties$7(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose$7(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose$7(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var CheckSquare = forwardRef(function (_ref, ref) {
  var _ref$color = _ref.color,
      color = _ref$color === void 0 ? 'currentColor' : _ref$color,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 24 : _ref$size,
      rest = _objectWithoutProperties$7(_ref, ["color", "size"]);

  return /*#__PURE__*/React.createElement("svg", _extends$7({
    ref: ref,
    xmlns: "http://www.w3.org/2000/svg",
    width: size,
    height: size,
    viewBox: "0 0 24 24",
    fill: "none",
    stroke: color,
    strokeWidth: "2",
    strokeLinecap: "round",
    strokeLinejoin: "round"
  }, rest), /*#__PURE__*/React.createElement("polyline", {
    points: "9 11 12 14 22 4"
  }), /*#__PURE__*/React.createElement("path", {
    d: "M21 12v7a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h11"
  }));
});
CheckSquare.propTypes = {
  color: PropTypes.string,
  size: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};
CheckSquare.displayName = 'CheckSquare';
var CheckSquare$1 = CheckSquare;

function _extends$6() { _extends$6 = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$6.apply(this, arguments); }

function _objectWithoutProperties$6(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose$6(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose$6(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var Check = forwardRef(function (_ref, ref) {
  var _ref$color = _ref.color,
      color = _ref$color === void 0 ? 'currentColor' : _ref$color,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 24 : _ref$size,
      rest = _objectWithoutProperties$6(_ref, ["color", "size"]);

  return /*#__PURE__*/React.createElement("svg", _extends$6({
    ref: ref,
    xmlns: "http://www.w3.org/2000/svg",
    width: size,
    height: size,
    viewBox: "0 0 24 24",
    fill: "none",
    stroke: color,
    strokeWidth: "2",
    strokeLinecap: "round",
    strokeLinejoin: "round"
  }, rest), /*#__PURE__*/React.createElement("polyline", {
    points: "20 6 9 17 4 12"
  }));
});
Check.propTypes = {
  color: PropTypes.string,
  size: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};
Check.displayName = 'Check';
var Check$1 = Check;

function _extends$5() { _extends$5 = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$5.apply(this, arguments); }

function _objectWithoutProperties$5(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose$5(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose$5(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var ExternalLink = forwardRef(function (_ref, ref) {
  var _ref$color = _ref.color,
      color = _ref$color === void 0 ? 'currentColor' : _ref$color,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 24 : _ref$size,
      rest = _objectWithoutProperties$5(_ref, ["color", "size"]);

  return /*#__PURE__*/React.createElement("svg", _extends$5({
    ref: ref,
    xmlns: "http://www.w3.org/2000/svg",
    width: size,
    height: size,
    viewBox: "0 0 24 24",
    fill: "none",
    stroke: color,
    strokeWidth: "2",
    strokeLinecap: "round",
    strokeLinejoin: "round"
  }, rest), /*#__PURE__*/React.createElement("path", {
    d: "M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
  }), /*#__PURE__*/React.createElement("polyline", {
    points: "15 3 21 3 21 9"
  }), /*#__PURE__*/React.createElement("line", {
    x1: "10",
    y1: "14",
    x2: "21",
    y2: "3"
  }));
});
ExternalLink.propTypes = {
  color: PropTypes.string,
  size: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};
ExternalLink.displayName = 'ExternalLink';
var ExternalLink$1 = ExternalLink;

function _extends$4() { _extends$4 = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$4.apply(this, arguments); }

function _objectWithoutProperties$4(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose$4(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose$4(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var Info = forwardRef(function (_ref, ref) {
  var _ref$color = _ref.color,
      color = _ref$color === void 0 ? 'currentColor' : _ref$color,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 24 : _ref$size,
      rest = _objectWithoutProperties$4(_ref, ["color", "size"]);

  return /*#__PURE__*/React.createElement("svg", _extends$4({
    ref: ref,
    xmlns: "http://www.w3.org/2000/svg",
    width: size,
    height: size,
    viewBox: "0 0 24 24",
    fill: "none",
    stroke: color,
    strokeWidth: "2",
    strokeLinecap: "round",
    strokeLinejoin: "round"
  }, rest), /*#__PURE__*/React.createElement("circle", {
    cx: "12",
    cy: "12",
    r: "10"
  }), /*#__PURE__*/React.createElement("line", {
    x1: "12",
    y1: "16",
    x2: "12",
    y2: "12"
  }), /*#__PURE__*/React.createElement("line", {
    x1: "12",
    y1: "8",
    x2: "12.01",
    y2: "8"
  }));
});
Info.propTypes = {
  color: PropTypes.string,
  size: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};
Info.displayName = 'Info';
var Info$1 = Info;

function _extends$3() { _extends$3 = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$3.apply(this, arguments); }

function _objectWithoutProperties$3(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose$3(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose$3(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var MoreHorizontal = forwardRef(function (_ref, ref) {
  var _ref$color = _ref.color,
      color = _ref$color === void 0 ? 'currentColor' : _ref$color,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 24 : _ref$size,
      rest = _objectWithoutProperties$3(_ref, ["color", "size"]);

  return /*#__PURE__*/React.createElement("svg", _extends$3({
    ref: ref,
    xmlns: "http://www.w3.org/2000/svg",
    width: size,
    height: size,
    viewBox: "0 0 24 24",
    fill: "none",
    stroke: color,
    strokeWidth: "2",
    strokeLinecap: "round",
    strokeLinejoin: "round"
  }, rest), /*#__PURE__*/React.createElement("circle", {
    cx: "12",
    cy: "12",
    r: "1"
  }), /*#__PURE__*/React.createElement("circle", {
    cx: "19",
    cy: "12",
    r: "1"
  }), /*#__PURE__*/React.createElement("circle", {
    cx: "5",
    cy: "12",
    r: "1"
  }));
});
MoreHorizontal.propTypes = {
  color: PropTypes.string,
  size: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};
MoreHorizontal.displayName = 'MoreHorizontal';
var MoreHorizontal$1 = MoreHorizontal;

function _extends$2() { _extends$2 = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$2.apply(this, arguments); }

function _objectWithoutProperties$2(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose$2(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose$2(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var Printer = forwardRef(function (_ref, ref) {
  var _ref$color = _ref.color,
      color = _ref$color === void 0 ? 'currentColor' : _ref$color,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 24 : _ref$size,
      rest = _objectWithoutProperties$2(_ref, ["color", "size"]);

  return /*#__PURE__*/React.createElement("svg", _extends$2({
    ref: ref,
    xmlns: "http://www.w3.org/2000/svg",
    width: size,
    height: size,
    viewBox: "0 0 24 24",
    fill: "none",
    stroke: color,
    strokeWidth: "2",
    strokeLinecap: "round",
    strokeLinejoin: "round"
  }, rest), /*#__PURE__*/React.createElement("polyline", {
    points: "6 9 6 2 18 2 18 9"
  }), /*#__PURE__*/React.createElement("path", {
    d: "M6 18H4a2 2 0 0 1-2-2v-5a2 2 0 0 1 2-2h16a2 2 0 0 1 2 2v5a2 2 0 0 1-2 2h-2"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "6",
    y: "14",
    width: "12",
    height: "8"
  }));
});
Printer.propTypes = {
  color: PropTypes.string,
  size: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};
Printer.displayName = 'Printer';
var Printer$1 = Printer;

function _extends$1() { _extends$1 = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$1.apply(this, arguments); }

function _objectWithoutProperties$1(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose$1(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose$1(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var Square = forwardRef(function (_ref, ref) {
  var _ref$color = _ref.color,
      color = _ref$color === void 0 ? 'currentColor' : _ref$color,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 24 : _ref$size,
      rest = _objectWithoutProperties$1(_ref, ["color", "size"]);

  return /*#__PURE__*/React.createElement("svg", _extends$1({
    ref: ref,
    xmlns: "http://www.w3.org/2000/svg",
    width: size,
    height: size,
    viewBox: "0 0 24 24",
    fill: "none",
    stroke: color,
    strokeWidth: "2",
    strokeLinecap: "round",
    strokeLinejoin: "round"
  }, rest), /*#__PURE__*/React.createElement("rect", {
    x: "3",
    y: "3",
    width: "18",
    height: "18",
    rx: "2",
    ry: "2"
  }));
});
Square.propTypes = {
  color: PropTypes.string,
  size: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};
Square.displayName = 'Square';
var Square$1 = Square;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var X = forwardRef(function (_ref, ref) {
  var _ref$color = _ref.color,
      color = _ref$color === void 0 ? 'currentColor' : _ref$color,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 24 : _ref$size,
      rest = _objectWithoutProperties(_ref, ["color", "size"]);

  return /*#__PURE__*/React.createElement("svg", _extends({
    ref: ref,
    xmlns: "http://www.w3.org/2000/svg",
    width: size,
    height: size,
    viewBox: "0 0 24 24",
    fill: "none",
    stroke: color,
    strokeWidth: "2",
    strokeLinecap: "round",
    strokeLinejoin: "round"
  }, rest), /*#__PURE__*/React.createElement("line", {
    x1: "18",
    y1: "6",
    x2: "6",
    y2: "18"
  }), /*#__PURE__*/React.createElement("line", {
    x1: "6",
    y1: "6",
    x2: "18",
    y2: "18"
  }));
});
X.propTypes = {
  color: PropTypes.string,
  size: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};
X.displayName = 'X';
var X$1 = X;

var PoolCard = function (_a) {
    var poolIcon = _a.poolIcon, poolTitle = _a.poolTitle, poolDescription = _a.poolDescription, poolStat = _a.poolStat, poolSubstat = _a.poolSubstat, poolState = _a.poolState, unitranche = _a.unitranche, props = __rest(_a, ["poolIcon", "poolTitle", "poolDescription", "poolStat", "poolSubstat", "poolState", "unitranche"]);
    return (React.createElement("div", { className: "pool-card--container" },
        React.createElement("div", { className: "pool--title" },
            React.createElement("div", { className: "item1" },
                " ",
                React.createElement("img", { src: poolIcon })),
            React.createElement("div", { className: "item2" }, poolTitle),
            React.createElement("div", { className: "item3" }, poolDescription)),
        React.createElement("div", { className: "pool--stats" },
            React.createElement("span", null, poolStat),
            React.createElement("span", null,
                poolSubstat,
                React.createElement(Info$1, { color: "#64748b" }))),
        React.createElement("div", { className: "pool--button" },
            React.createElement(Button, { label: unitranche ? "Unitranche" : "Multitranche", onClick: function () { } }),
            React.createElement(Button, __assign({ disabled: poolState == "closed" || poolState == "full", backgroundColor: poolState == "yield" ? "#1ab98b" : "#d70b48", primary: true, label: poolState }, props)))));
};

var css_248z$5 = ".status {\n  font-family: \"Inter\", sans-serif;\n  align-items: center;\n  border-radius: 50px;\n  text-align: center;\n  font-size: 14px;\n  padding-left: 10px;\n  padding-right: 10px;\n  padding-top: 5px;\n  padding-bottom: 5px;\n}\n\n.status--without--icon {\n  width: auto;\n  height: auto;\n}\n\n.status--with--icon {\n  width: -moz-min-content;\n  width: min-content;\n  height: -moz-min-content;\n  height: min-content;\n  margin: 0px;\n  display: flex;\n  justify-content: center;\n  font-weight: 400;\n}\n.status--with--icon svg {\n  padding: 0px;\n  margin: 0px;\n  width: 15px;\n  height: 15px;\n  margin-right: 8px;\n}\n\n.progressbar {\n  font-family: \"Inter\", sans-serif;\n  margin-top: 5px;\n  background-color: rgba(5, 17, 19, 0.2);\n  border-radius: 5px;\n  height: 10px;\n  width: 177px;\n}\n\n.progressbar--done {\n  font-family: \"Inter\", sans-serif;\n  border-radius: 20px;\n  color: #fff;\n  display: flex;\n  align-items: center;\n  justify-content: center;\n  height: 100%;\n  width: 0;\n  opacity: 0;\n  transition: 1s ease 0.3s;\n}\n\n.slider--container {\n  width: 100%;\n}\n.slider--container .slider {\n  z-index: 0;\n  -webkit-appearance: none;\n  width: 100%;\n  height: 10px;\n  border-radius: 5px;\n  background: linear-gradient(to right, rgba(239, 68, 68, 0.5), rgba(122, 109, 87, 0.15), rgba(5, 150, 105, 0.5));\n  outline: none;\n  opacity: 1;\n  transition: opacity 0.15s ease-in-out;\n}\n.slider--container .slider::-webkit-slider-thumb {\n  -webkit-appearance: none;\n  appearance: none;\n  width: 16px;\n  height: 16px;\n  border: 2px solid #81cfdb;\n  border-radius: 50%;\n  background: white;\n  cursor: pointer;\n}\n.slider--container .slider::-moz-range-thumb {\n  -moz-appearance: none;\n       appearance: none;\n  width: 16px;\n  height: 16px;\n  border: 2px solid #81cfdb;\n  border-radius: 50%;\n  background: white;\n  cursor: pointer;\n}\n\n.activity--box {\n  margin-top: 16px;\n  width: -moz-max-content;\n  width: max-content;\n  height: 72px;\n  background-color: #fff;\n  border-radius: 5px;\n  border: 1px solid #e2e8f0;\n  display: flex;\n  flex-direction: row;\n}\n.activity--box .status--box {\n  width: 63px;\n  height: 72px;\n  display: flex;\n  align-items: center;\n  justify-content: center;\n}\n.activity--box .detail--box {\n  width: 1039px;\n  height: 71px;\n  display: flex;\n  display: flex;\n  flex-direction: column;\n  justify-content: center;\n}\n.activity--box .detail--box .content :nth-child(1) {\n  color: #0d8489;\n  font-weight: 500;\n  margin-right: 3px;\n}\n.activity--box .detail--box .content :nth-child(2) {\n  color: #000000;\n  font-weight: 500;\n  margin-right: 3px;\n}\n.activity--box .detail--box :nth-child(2) {\n  color: #94a3b8;\n  font-weight: 400;\n  font-size: 13px;\n}";
styleInject(css_248z$5);

// Generated with util/create-component.js
var colorHash = {
    active: {
        bgcolor: '#D1FAE5',
        color: '#059669'
    },
    change: {
        bgcolor: '#FEE2E2',
        color: '#EF4444'
    },
    review: {
        bgcolor: '#FEF3C7',
        color: '#F59E0B'
    },
    completed: {
        bgcolor: '#10B981',
        color: '#ECFDF5'
    }
};
function StatusWidget(_a) {
    var type = _a.type, icon = _a.icon, showIcon = _a.showIcon;
    var _b = useState({}), style = _b[0], setStyle = _b[1];
    var mode = showIcon ? 'status--with--icon' : 'status--without--icon';
    useEffect(function () {
        switch (type.toLowerCase()) {
            case 'up':
                setStyle({
                    opacity: 1,
                    backgroundColor: colorHash.active.bgcolor,
                    color: colorHash.active.color
                });
                break;
            case 'down':
                setStyle({
                    opacity: 1,
                    backgroundColor: colorHash.change.bgcolor,
                    color: colorHash.change.color
                });
                break;
            case 'change' :
                setStyle({
                    opacity: 1,
                    backgroundColor: colorHash.change.bgcolor,
                    color: colorHash.change.color
                });
                break;
            case 'review':
                setStyle({
                    opacity: 1,
                    backgroundColor: colorHash.review.bgcolor,
                    color: colorHash.review.color
                });
                break;
            case 'completed':
                setStyle({
                    opacity: 1,
                    backgroundColor: colorHash.completed.bgcolor,
                    color: colorHash.completed.color
                });
                break;
            default:
                setStyle({
                    opacity: 1,
                    backgroundColor: colorHash.active.bgcolor,
                    color: colorHash.active.color
                });
                break;
        }
    }, [type]);
    return (React.createElement("div", { className: ['status', mode].join(' '), style: style },
        showIcon ? icon : '',
        type));
}
function ProgressBar(_a) {
    var progress = _a.progress, color = _a.color, showText = _a.showText;
    var _b = React.useState({}), style = _b[0], setStyle = _b[1];
    setTimeout(function () {
        var newStyle = {
            opacity: 1,
            width: "".concat(progress, "%"),
            backgroundColor: color
        };
        setStyle(newStyle);
    }, 100);
    return (React.createElement("div", null,
        showText ? React.createElement("span", null,
            "Overall Progress: ",
            progress,
            "%") : '',
        React.createElement("div", { className: "progressbar" },
            React.createElement("div", { className: "progressbar--done", style: style }))));
}
var MetricRange = function (_a) {
    var value = _a.value;
    return (React.createElement(React.Fragment, null,
        React.createElement("div", { className: "slider--container" },
            React.createElement("input", { className: "slider", type: "range", min: "1", max: "100", value: value }))));
};
var Activity = function (_a) {
    var user = _a.user, activity = _a.activity, status = _a.status, date = _a.date;
    return (React.createElement("div", { className: "activity--box" },
        React.createElement("div", { className: "status--box" }, status ? React.createElement(Check$1, null) : React.createElement(X$1, null)),
        React.createElement("div", { className: "detail--box" },
            React.createElement("div", { className: "content" },
                React.createElement("span", null, user),
                React.createElement("span", null, activity)),
            React.createElement("span", null, date))));
};

var css_248z$4 = ".table {\n  font-family: \"inter\", sans-serif;\n  margin-top: 30px;\n  /* top-left border-radius */\n  /* top-right border-radius \n\n  /* bottom-left border-radius */\n  /* bottom-right border-radius */\n}\n.table table {\n  border-collapse: separate;\n  border-spacing: 0;\n  min-width: 350px;\n  width: 100%;\n}\n.table table tr,\n.table table tr td {\n  border-bottom: 1px solid #e2e8f0;\n  padding: 25px;\n  font-weight: bold;\n}\n.table table tr th:first-child,\n.table table tr td:first-child {\n  border-left: 1px solid #e2e8f0;\n}\n.table table tr th:first-child,\n.table table tr td:first-child {\n  border-left: 1px solid #e2e8f0;\n}\n.table table tr th {\n  background: #f8fafc;\n  border-bottom: 1px solid #e2e8f0;\n  text-align: left;\n  padding: 10px 25px 10px 25px;\n  border-top: solid 1px #e2e8f0;\n  font-size: small;\n  font-weight: bold;\n  color: #475569;\n}\n.table table tr:first-child th:first-child {\n  border-top-left-radius: 10px;\n}\n.table table tr:last-child td:first-child {\n  border-bottom-left-radius: 10px;\n}\n.table table tr:first-child th:last-child {\n  border-top-right-radius: 10px;\n  border-right: 1px solid #e2e8f0;\n}\n.table table tr:last-child td:last-child {\n  border-bottom-right-radius: 10px;\n  border-right: 1px solid #e2e8f0;\n}\n.table table tr td:last-child {\n  border-right: 1px solid #e2e8f0;\n}\n.table table tr td {\n  background: #ffffff;\n  font-size: 14px;\n}\n.table .hit--tab {\n  font-size: 11px;\n  font-weight: normal;\n  width: 500px;\n}\n.table .user--container {\n  width: 100%;\n}\n.table .persons--tab {\n  width: 400px;\n}\n.table .svg--icon {\n  display: flex;\n  align-items: center;\n  gap: 6px;\n}\n.table .svg--icon .selected {\n  font-weight: bold;\n}\n.table .svg--icon .not--selected {\n  font-weight: normal;\n  color: #cbd5e1;\n}\n.table .source--tab {\n  gap: 6px;\n}\n.table .source--tab a {\n  font-weight: bold;\n  font-size: 13px;\n  color: #000;\n  text-decoration: none;\n}\n.table .source--tab :nth-child(2) {\n  position: relative;\n  bottom: -5px;\n}\n.table .persons--container {\n  display: flex;\n  width: 100%;\n  align-items: center;\n  gap: 8px;\n}\n.table .persons--container .images {\n  width: 32px;\n  height: 32px;\n}\n.table .email--container {\n  font-weight: normal;\n  color: #94a3b8;\n}\n.table .sent .sent--done {\n  display: flex;\n  align-items: center;\n  justify-content: center;\n  width: 55px;\n  height: 25px;\n  border-radius: 50px;\n  background-color: #d1fae5;\n  color: #059669;\n  text-align: center;\n  font-weight: normal;\n}\n.table .sent .sent--not--done {\n  display: flex;\n  align-items: center;\n  justify-content: center;\n  width: 83px;\n  height: 25px;\n  border-radius: 50px;\n  background-color: #fee2e2;\n  color: #ef4444;\n  text-align: center;\n  font-weight: normal;\n}\n.table .done .complete {\n  display: flex;\n  align-items: center;\n  justify-content: center;\n  width: 87px;\n  height: 25px;\n  border-radius: 50px;\n  background-color: #fef3c7;\n  color: #f59e0b;\n  text-align: center;\n  font-weight: normal;\n}\n.table .done .not--done {\n  display: flex;\n  align-items: center;\n  justify-content: center;\n  width: 97px;\n  height: 25px;\n  border-radius: 50px;\n  background-color: #10b981;\n  color: #ecfdf5;\n  text-align: center;\n  font-weight: normal;\n}\n.table .user--container {\n  display: flex;\n  gap: 5px;\n}\n.table .user--container .flex--box {\n  display: flex;\n  flex-direction: column;\n  gap: 2px;\n}\n.table .user--container .flex--box :nth-child(2) {\n  font-weight: normal;\n  color: #94a3b8;\n  font-size: 12px;\n}";
styleInject(css_248z$4);

// Generated with util/create-component.js
var ReceiptTable = function (_a) {
    var receiptTableData = _a.receiptTableData;
    return (React.createElement("div", { className: "table" },
        React.createElement("table", null,
            React.createElement("tbody", null,
                React.createElement("tr", { className: "title--row" }, [
                    'Transaction Date',
                    'Amount',
                    'Remaining Balance',
                    'Payment Method',
                    'Options',
                ].map(function (item, index) {
                    return (React.createElement("th", { key: index, className: "title--headder" }, item));
                })),
                receiptTableData.map(function (item, index) {
                    return (React.createElement("tr", { key: index, className: "table--content" },
                        React.createElement("td", { className: "app--table--items" }, item.transactionDate),
                        React.createElement("td", { className: "app--table--items" }, item.amount),
                        React.createElement("td", { className: "app--table--items" }, item.remainingBalance),
                        React.createElement("td", { className: "app--table--items" }, item.paymentMethod),
                        React.createElement("td", { className: "app--table--items" },
                            React.createElement("a", { href: item.url, "attributes-list": true, download: true },
                                React.createElement(Printer$1, { color: "#65768D" })))));
                })))));
};
var RepaymentTable = function (_a) {
    var repaymentTableData = _a.repaymentTableData;
    return (React.createElement("div", { className: "table" },
        React.createElement("table", null,
            React.createElement("tbody", null,
                React.createElement("tr", { className: "title--row" }, ['Payment', 'Amount', 'Due Date', 'status'].map(function (item, index) {
                    return (React.createElement("th", { key: index, className: "title--headder" }, item));
                })),
                repaymentTableData.map(function (item, index) {
                    return (React.createElement("tr", { key: index, className: "table--content" },
                        React.createElement("td", { className: "app--table--items" }, item.paymentId),
                        React.createElement("td", { className: "app--table--items" }, item.amount),
                        React.createElement("td", { className: "app--table--items" }, item.dueDate),
                        React.createElement("td", { className: "app--table--items" }, item.status)));
                })))));
};
var ApplicationTable = function (_a) {
    var applicationTableData = _a.applicationTableData;
    return (React.createElement("div", { className: "table" },
        React.createElement("table", null,
            React.createElement("tbody", null,
                React.createElement("tr", { className: "title--row" }, [
                    'Application ID',
                    'Product',
                    'Date From',
                    'Date To',
                    'Importer ID',
                    'Importer Name',
                    'Status',
                ].map(function (item, index) {
                    return (React.createElement("th", { key: index, className: "title--headder" }, item));
                })),
                applicationTableData.map(function (item, index) {
                    return (React.createElement("tr", { key: index, className: "table--content" },
                        React.createElement("td", { className: "app--table--items" }, item.applicationId),
                        React.createElement("td", { className: "app--table--items" }, item.product),
                        React.createElement("td", { className: "app--table--items" }, item.dateFrom),
                        React.createElement("td", { className: "app--table--items" }, item.dateTo),
                        React.createElement("td", { className: "app--table--items" }, item.importerId),
                        React.createElement("td", { className: "app--table--items" }, item.importerName),
                        React.createElement("td", { className: "app--table--items" }, item.status)));
                })))));
};
var EntityTable = function (_a) {
    var entityTableData = _a.entityTableData;
    return (React.createElement("div", { className: "table" },
        React.createElement("table", null,
            React.createElement("tbody", null,
                React.createElement("tr", { className: "status--title" }, ['Title', 'Issuance Date', 'Maturity Date', 'Facility Amount'].map(function (item, index) {
                    return (React.createElement("th", { key: index, className: "status--title--items" }, item));
                })),
                entityTableData.map(function (item, index) {
                    return (React.createElement("tr", { className: "table--Content", key: index },
                        React.createElement("td", { className: "persons--tab" }, item.productTitle),
                        React.createElement("td", { className: "details--tab" }, item.issuanceDate),
                        React.createElement("td", { className: "source--tab" }, item.maturityDate),
                        React.createElement("td", { className: "status--tab" }, item.facilityAmount)));
                })))));
};
var CompanyTable = function (_a) {
    var companyTableData = _a.companyTableData;
    return (React.createElement("div", { className: "table" },
        React.createElement("table", null,
            React.createElement("tbody", null,
                React.createElement("tr", { className: "status--title" }, ['Persons', 'Details', 'Source', 'Status'].map(function (item, index) {
                    return (React.createElement("th", { key: index, className: "statusTitleItems" }, item));
                })),
                companyTableData.map(function (item, index) {
                    return (React.createElement("tr", { className: "tableContent", key: index },
                        React.createElement("td", { className: "details--tab" }, item.persons),
                        React.createElement("td", { className: "persons--tab" }, item.details),
                        React.createElement("td", { className: "source--tab" },
                            React.createElement("a", { href: item.source }, item.source),
                            "\u00A0\u00A0",
                            React.createElement(ExternalLink$1, { color: "#0BD7A4" })),
                        React.createElement("td", { className: "status--tab" }, item.status ? (React.createElement("div", { className: "svg--icon" },
                            React.createElement(CheckSquare$1, { color: "#0BD7A4" }),
                            React.createElement("span", { className: "selected" }, "Verified"))) : (React.createElement("div", { className: "svg--icon" },
                            React.createElement(Square$1, { color: "#CBD5E1" }),
                            React.createElement("span", { className: "not--selected" }, "Verified"))))));
                })))));
};
var ScreeningTable = function (_a) {
    var screeningTableData = _a.screeningTableData;
    return (React.createElement("div", { className: "table" },
        React.createElement("table", null,
            React.createElement("tbody", null,
                React.createElement("tr", { className: "status--title" }, ['Persons', 'Hit', 'Source', 'Options'].map(function (item, index) {
                    return (React.createElement("th", { key: index, className: "statusTitleItems" }, item));
                })),
                screeningTableData.map(function (item, index) {
                    return (React.createElement("tr", { className: "tableContent", key: index },
                        React.createElement("td", { className: "details--tab" }, item.persons),
                        React.createElement("td", { className: "hit--tab" }, item.Hit),
                        React.createElement("td", { className: "source--tab" },
                            React.createElement("a", { href: item.Source }, item.Source),
                            "\u00A0\u00A0",
                            React.createElement(ExternalLink$1, { color: "#0BD7A4" })),
                        React.createElement("td", { className: "status--tab" },
                            React.createElement("div", { className: "svg--icon" },
                                React.createElement(MoreHorizontal$1, { color: "#64748b" })))));
                })))));
};
var PersonsTable = function (_a) {
    var personsTableData = _a.personsTableData;
    return (React.createElement("div", { className: 'table' },
        React.createElement("table", null,
            React.createElement("tbody", null,
                React.createElement("tr", { className: "status--title" }, [
                    'Persons',
                    'Email',
                    'Position',
                    'Verification Status',
                    'Status Verification',
                    'Options',
                ].map(function (item, index) {
                    return React.createElement("th", { key: index }, item);
                })),
                personsTableData.map(function (item, index) {
                    return (React.createElement("tr", { key: index },
                        React.createElement("td", { className: "persons--container", id: "assignee" },
                            React.createElement("img", { className: "images", src: item.image, key: index, alt: "", width: "35%", height: "35%" }),
                            React.createElement("div", { className: "name--container" },
                                React.createElement("span", null, item.name))),
                        React.createElement("td", { className: "email--container" },
                            React.createElement("span", null, item.email)),
                        React.createElement("td", { id: "assignor" },
                            React.createElement("span", null, item.position)),
                        React.createElement("td", { className: "sent" }, item.isSent ? (React.createElement("div", { className: "sent--done" }, "Sent")) : (React.createElement("div", { className: "sent--not--done" }, "Not Sent"))),
                        React.createElement("td", { className: "done" }, item.isDone ? (React.createElement("div", { className: "not--done" }, "Completed")) : (React.createElement("div", { className: "complete" }, "Not Done"))),
                        React.createElement("td", null,
                            React.createElement(MoreHorizontal$1, { color: "#64748b" }))));
                })))));
};
var ResearchTable = function (_a) {
    var researchTableData = _a.researchTableData;
    return (React.createElement("div", { className: "table" },
        React.createElement("table", null,
            React.createElement("tbody", null,
                React.createElement("tr", { className: "status--title" }, ['Persons', 'Hit', 'Source', 'Options'].map(function (item, index) {
                    return React.createElement("th", { key: index }, item);
                })),
                researchTableData.map(function (item, index) {
                    return (React.createElement("tr", { key: index },
                        React.createElement("td", { className: "persons--container", id: "assignee" },
                            React.createElement("img", { className: "images", src: item.image, key: index, alt: "", width: "35%", height: "35%" }),
                            React.createElement("div", { className: "name--container" },
                                React.createElement("span", null, item.name))),
                        React.createElement("td", { className: "hit--tab" }, item.Hit),
                        React.createElement("td", { className: "source--tab" },
                            React.createElement("a", { href: item.Source }, item.Source),
                            "\u00A0\u00A0",
                            React.createElement(ExternalLink$1, { color: "#0BD7A4" })),
                        React.createElement("td", { className: "status--tab" },
                            React.createElement("div", { className: "svg--icon" },
                                React.createElement(MoreHorizontal$1, { color: "#64748b" })))));
                })))));
};
var SignOffTable = function (_a) {
    var signOffTableData = _a.signOffTableData;
    return (React.createElement("div", { className: "table" },
        React.createElement("table", null,
            React.createElement("tbody", null,
                React.createElement("tr", null, ['Name', 'Position', 'Status', 'Status'].map(function (item, index) {
                    return React.createElement("th", { key: index }, item);
                })),
                signOffTableData.map(function (item, index) {
                    return (React.createElement("tr", { key: index },
                        React.createElement("td", { className: "user--container" },
                            React.createElement("img", { className: "images", src: item.image, key: index, alt: "", width: "32px", height: "32px" }),
                            React.createElement("div", { className: "flex--box" },
                                React.createElement("span", null, item.name),
                                React.createElement("span", null, item.email))),
                        React.createElement("td", null, item.position),
                        React.createElement("td", null, item.statusSeen ? (React.createElement("div", { className: "svg--icon" },
                            React.createElement(CheckSquare$1, { color: "#0BD7A4" }),
                            React.createElement("span", { className: "selected" }, "Seen"))) : (React.createElement("div", { className: "svg--icon" },
                            React.createElement(Square$1, { color: "#CBD5E1" }),
                            React.createElement("span", { className: "not--selected" }, "Seen")))),
                        React.createElement("td", null, item.statusSigned ? (React.createElement("div", { className: "svg--icon" },
                            React.createElement(CheckSquare$1, { color: "#0BD7A4" }),
                            React.createElement("span", { className: "selected" }, "Signed"))) : (React.createElement("div", { className: "svg--icon" },
                            React.createElement(Square$1, { color: "#CBD5E1" }),
                            React.createElement("span", { className: "not--selected" }, "Signed"))))));
                })))));
};

var css_248z$3 = ".deleteModal {\n  overflow-y: hidden;\n  overflow-x: hidden;\n  font-family: \"inter\", sans-serif;\n}\n.deleteModal.active {\n  box-shadow: 3px 18px 36px 100vw rgba(93, 96, 99, 0.42);\n  -webkit-box-shadow: 3px 18px 36px 100vw rgba(93, 96, 99, 0.42);\n  -moz-box-shadow: 3px 18px 36px 100vw rgba(93, 96, 99, 0.42);\n  width: 27vw;\n  height: -moz-fit-content;\n  height: fit-content;\n  left: 35vw;\n  top: 12vw;\n  background-color: #fff;\n  position: fixed;\n  z-index: 99;\n  border-radius: 16px;\n  animation: bounce-in 0.5s ease;\n  -moz-animation: bounce-in 0.5s ease;\n  /* Firefox */\n  -webkit-animation: bounce-in 0.5s ease;\n  /* Safari and Chrome */\n}\n@keyframes bounce-in {\n  0% {\n    opacity: 0;\n    transform: scale(0.3);\n  }\n  50% {\n    opacity: 1;\n    transform: scale(1.05);\n  }\n  100% {\n    transform: scale(1);\n  }\n}\n.deleteModal .modalMenue {\n  display: flex;\n  flex-direction: column;\n  gap: 25px;\n}\n.deleteModal .modalMenue .close {\n  position: relative;\n  color: rgb(78, 78, 78);\n  top: 20px;\n  left: 20px;\n}\n.deleteModal .modalMenue .header {\n  width: 100%;\n  padding: auto;\n  margin: auto;\n  padding-top: 5%;\n  display: flex;\n  text-align: center;\n  flex-direction: column;\n  font-size: 2vw;\n  font-weight: 800;\n  color: black;\n  /* Prevent shrinking of header */\n  min-height: 60px;\n  min-width: 150px;\n}\n.deleteModal .modalMenue .header span {\n  text-align: center;\n  padding: 15px;\n  margin-inline: 20px;\n  font-size: 1vw;\n  font-weight: normal;\n  color: gray;\n}\n.deleteModal .modalMenue .form {\n  width: 70%;\n  gap: 10px;\n  padding-bottom: 40px;\n  margin: auto;\n  display: flex;\n  flex-direction: column;\n  justify-content: space-around;\n  /* Prevent shrinking of form */\n  min-height: 100px;\n  min-width: 200px;\n}\n.deleteModal .modalMenue .form .button {\n  border: 0;\n  color: white;\n  background-color: #d70b48;\n  border-radius: 5px;\n  height: 40px;\n  display: flex;\n  align-items: center;\n  justify-content: center;\n  width: 100%;\n  /* Prevent shrinking of buttons */\n  min-height: 40px;\n  min-width: 100px;\n}\n.deleteModal .modalMenue .form .buttonEdit {\n  border: 2px #1d4ed8 solid;\n  color: #1d4ed8;\n  background-color: white;\n  border-radius: 5px;\n  height: 40px;\n  display: flex;\n  align-items: center;\n  justify-content: center;\n  width: 100%;\n  font-weight: bold;\n}\n.deleteModal .modalMenue .form .label {\n  display: flex;\n  justify-content: flex-start;\n  color: gray;\n  font-size: 15px;\n  text-transform: uppercase;\n  font-weight: 600;\n}";
styleInject(css_248z$3);

// Generated with util/create-component.js
var DeleteModal = function (_a) {
    var title = _a.title, isOpen = _a.isOpen, onClose = _a.onClose, onDelete = _a.onDelete;
    return (React.createElement("div", { className: 'deleteModal ' + (isOpen && 'active') }, isOpen && (React.createElement("div", { className: "modalMenue" },
        React.createElement("div", { onClick: function () {
                onClose();
            }, className: "close" },
            React.createElement(X$1, { color: "#64748b" })),
        React.createElement("div", { className: "header" },
            title,
            React.createElement("span", { className: "subtitle" }, "Are you sure you want to delete this item? This is an irreversible action and the data associated with this entity will be deleted.")),
        React.createElement("div", { className: "form" },
            React.createElement("button", { className: "button", onClick: function () {
                    onDelete();
                } }, "Delete"),
            React.createElement("button", { className: "buttonEdit", onClick: function () {
                    onClose();
                } }, "Keep it for now"))))));
};

var css_248z$2 = ".entityWrapper {\n  display: flex;\n  flex-direction: column;\n  padding: 20px 0px 20px 16px;\n  width: -moz-fit-content;\n  width: fit-content;\n  height: -moz-fit-content;\n  height: fit-content;\n  border: 1px solid #e2e8f0;\n  border-radius: 16px;\n  margin-right: 24px;\n  margin-bottom: 24px;\n}\n.entityWrapper :hover {\n  box-shadow: initial;\n}\n.entityWrapper .entityTopContent {\n  display: flex;\n  padding-bottom: 16px;\n  border-bottom: 1px solid #e2e8f0;\n  width: 300px;\n}\n.entityWrapper .entityTopContent .entityImageContainer {\n  width: 80px;\n  height: 80px;\n  contain: content;\n  border-radius: 16px;\n}\n.entityWrapper .entityTopContent .entityTitleBox {\n  gap: 5px;\n  padding: 16px 45px 16px 18px;\n  display: flex;\n  flex-direction: column;\n}\n.entityWrapper .entityTopContent .entityTitleBox :nth-child(1) {\n  font-style: Work Sans;\n  font-size: 24px;\n  color: #003c6e;\n  font-weight: bold;\n  width: -moz-max-content;\n  width: max-content;\n}\n.entityWrapper .entityTopContent .entityTitleBox :nth-child(2) {\n  font-style: Work Sans;\n  font-size: 10px;\n  color: #64748b;\n  font-weight: bold;\n}\n.entityWrapper .entityTopContent .optionsContainer {\n  margin-left: auto;\n}\n.entityWrapper .entityTopContent .modalContainer {\n  position: absolute;\n}\n.entityWrapper .entityTopContent .modalContainer.active {\n  width: 148px;\n  height: 192px;\n  background-color: #fff;\n  position: absolute;\n  z-index: 99;\n  border-radius: 16px;\n  box-shadow: 3px 18px 36px 1px rgba(100, 116, 139, 0.13);\n  -webkit-box-shadow: 3px 18px 36px 1px rgba(100, 116, 139, 0.13);\n  -moz-box-shadow: 3px 18px 36px 1px rgba(100, 116, 139, 0.13);\n}\n.entityWrapper .entityTopContent .modal .iconEntityOption {\n  opacity: 0.7;\n}\n.entityWrapper .entityTopContent .modal .iconEntityOption.active {\n  opacity: 1;\n}\n.entityWrapper .entityTopContent .modal .modalMenue {\n  display: flex;\n  flex-direction: column;\n  justify-content: space-around;\n  padding: 24px;\n  gap: 20px;\n}\n.entityWrapper .entityTopContent .modal .modalMenue :nth-child(1) {\n  display: flex;\n  gap: 13px;\n  align-items: center;\n}\n.entityWrapper .entityTopContent .modal .modalMenue :nth-child(2) {\n  display: flex;\n  gap: 16px;\n  align-items: center;\n}\n.entityWrapper .entityTopContent .modal .modalMenue :nth-child(4) {\n  display: flex;\n  gap: 15px;\n  align-items: center;\n}\n.entityWrapper .entityTopContent .modal .modalMenue :nth-child(3) {\n  width: 100%;\n  height: 1px;\n  background-color: rgba(13, 132, 137, 0.3);\n}\n.entityWrapper .bottomContent {\n  display: flex;\n  flex-direction: row;\n}\n.entityWrapper .bottomContent :nth-child(3) {\n  padding-right: 15px;\n}\n.entityWrapper .bottomContent * {\n  display: flex;\n  gap: 4px;\n  flex-direction: column;\n  padding: 16px 40px 0px 0px;\n}\n.entityWrapper .bottomContent * :nth-child(1) {\n  font-weight: 900;\n  color: #64748b;\n  font-size: 10px;\n  padding: 0px;\n}\n.entityWrapper .bottomContent * :nth-child(2) {\n  font-weight: 900;\n  color: #003c6e;\n  font-size: 16px;\n  padding: 0px;\n}";
styleInject(css_248z$2);

// Generated with util/create-component.js
function Entity(_a) {
    var entityData = _a.entityData, modal = _a.modal;
    return (React.createElement("div", { className: 'entityWrapper' },
        React.createElement("div", { className: "entityTopContent" },
            React.createElement("div", { className: "entityImageContainer" },
                React.createElement("img", { className: "images", src: entityData.image, alt: "", width: "100%", height: "100%" })),
            React.createElement("div", { className: "entityTitleBox" },
                React.createElement("span", null, entityData.title),
                React.createElement("span", null, "55-NB")),
            React.createElement("div", { className: "optionsContainer" }, modal)),
        React.createElement("div", { className: "bottomContent" },
            React.createElement("div", { className: "registrationContainer" },
                React.createElement("span", null, "REGISTRATION NUMBER"),
                React.createElement("span", null, entityData.registrationNo)),
            React.createElement("div", { className: "productContainer" },
                React.createElement("span", null, "PRODUCTS"),
                React.createElement("span", null, entityData.products)),
            React.createElement("div", { className: "applicationsContainer" },
                React.createElement("span", null, "APPLICATIONS"),
                React.createElement("span", null, entityData.applications)))));
}

var css_248z$1 = ".small--loan--card {\n  align-items: center;\n  background-color: white;\n  width: auto;\n  height: 80px;\n  display: flex;\n  flex-direction: row;\n  justify-content: space-between;\n  padding: 10px 50px 10px 50px;\n  gap: 20px;\n  border-radius: 10px;\n  box-shadow: 0px 0px 10px 0px rgba(0, 0, 0, 0.1);\n}\n.small--loan--card span {\n  font-family: \"inter\", sans-serif;\n  color: #65768d;\n}\n.small--loan--card .left--container {\n  font-size: 20px;\n  font-weight: 600;\n}\n.small--loan--card .right--container {\n  width: -moz-fit-content;\n  width: fit-content;\n  align-items: center;\n  display: flex;\n  gap: 5px;\n  flex-direction: column;\n}\n.small--loan--card .right--container span:nth-child(2) {\n  color: #b6c1d0;\n  font-weight: 100;\n}\n\n.large--loan--card {\n  font-family: \"inter\", sans-serif;\n  align-items: start;\n  background-color: white;\n  width: auto;\n  height: 180px;\n  display: flex;\n  flex-direction: column;\n  justify-content: center;\n  padding: 10px 50px 10px 50px;\n  gap: 10px;\n  border-radius: 10px;\n  box-shadow: 0px 0px 10px 0px rgba(0, 0, 0, 0.1);\n}\n.large--loan--card span {\n  color: #65768d;\n  font-size: 18px;\n}\n.large--loan--card .loan--card--container {\n  display: flex;\n  flex-direction: row;\n  gap: 5px;\n  align-items: baseline;\n}\n.large--loan--card .loan--card--container span {\n  color: #003c6e;\n  font-size: 30px;\n  font-weight: 800;\n}\n.large--loan--card .loan--card--container p {\n  color: #b6c1d0;\n  font-size: 14px;\n}\n\n.payment--success--card {\n  font-family: \"inter\", sans-serif;\n  background-color: white;\n  box-shadow: 0px 0px 10px 0px rgba(0, 0, 0, 0.1);\n  border-radius: 20px;\n  display: flex;\n  flex-direction: column;\n  align-items: center;\n  justify-content: center;\n  gap: 10px;\n  width: 100%;\n  padding: 5% 5%;\n}\n.payment--success--card .content--container {\n  padding-top: 30px;\n  display: flex;\n  flex-direction: row;\n  width: 60%;\n  gap: 20px;\n  align-items: center;\n  justify-content: center;\n}\n.payment--success--card .content--container .content--left {\n  gap: 20px;\n  display: flex;\n  flex-direction: column;\n  align-items: start;\n}\n.payment--success--card .content--container .content--right {\n  gap: 20px;\n  display: flex;\n  flex-direction: column;\n  align-items: start;\n}\n.payment--success--card .two--button--container {\n  padding-top: 30px;\n  display: flex;\n  justify-content: center;\n  gap: 40px;\n  width: 100%;\n  flex-direction: row;\n}\n.payment--success--card .two--button--container .button--container {\n  width: 100%;\n}";
styleInject(css_248z$1);

// Generated with util/create-component.js
var rightContainer = function (type) {
    switch (type.toUpperCase()) {
        case 'INTEREST_RATE':
            return React.createElement("span", { className: "right--container" }, "Interest Rate");
        case 'INTEREST_ACCRUED':
            return (React.createElement("div", { className: "right--container" },
                React.createElement("span", null, "Interest Accrued"),
                React.createElement("span", null, "(USD-StableCoin)")));
        case 'TOTAL_REPAYMENT':
            return (React.createElement("div", { className: "right--container" },
                React.createElement("span", null, "Total Repayments"),
                React.createElement("span", null, "(USD-StableCoin)")));
        case 'DUE_DATE':
            return React.createElement("span", { className: "right--container" }, "Next Instalment Due Date");
        default:
            return React.createElement(React.Fragment, null);
    }
};
var SmallLoanCard = function (_a) {
    var type = _a.type, value = _a.value;
    return (React.createElement("div", { "data-testid": "LoanCard", className: "small--loan--card" },
        React.createElement("span", { className: "left--container" }, value),
        rightContainer(type)));
};
var LargeLoanCard = function (_a) {
    var type = _a.type, value = _a.value;
    return (React.createElement("div", { "data-testid": "LoanCard", className: "large--loan--card" },
        React.createElement("div", { className: "loan--card--container" },
            React.createElement("span", null, value),
            React.createElement("p", null, "USD-Stablecoin")),
        React.createElement("span", null, type == 'LOAN_AMT' ? 'Loan Amount' : 'Outstanding Balance')));
};
var PaymentSuccessCard = function (_a) {
    var onPrint = _a.onPrint, onBack = _a.onBack, paymentDetails = _a.paymentDetails;
    return (React.createElement("div", { className: "payment--success--card" },
        React.createElement("h1", null, "Payment Successful!"),
        React.createElement("span", null, "Your payment transaction was successful. Thanks for using Tariala."),
        React.createElement("div", { className: "content--container" },
            React.createElement("div", { className: "content--left" },
                React.createElement("span", null, "Payment Amount"),
                React.createElement("span", null, "Payment Method"),
                React.createElement("span", null, "Transaction Date")),
            React.createElement("div", { className: "content--right" },
                React.createElement("span", null, paymentDetails.amount),
                React.createElement("span", null, paymentDetails.method),
                React.createElement("span", null, paymentDetails.date))),
        React.createElement("div", { className: "two--button--container" },
            React.createElement(Button, { label: "PRINT RECEIPT", onClick: function () {
                    onPrint();
                } }),
            React.createElement(Button, { label: "BACK HOME", onClick: function () {
                    onBack();
                } }))));
};

var css_248z = ".profile--address--card {\n  font-family: \"inter\", sans-serif;\n  background-color: white;\n  width: 100%;\n  height: auto;\n  display: flex;\n  flex-direction: column;\n  justify-content: space-between;\n  border-radius: 10px;\n  box-shadow: 0px 0px 10px 0px rgba(0, 0, 0, 0.1);\n}\n.profile--address--card .heading {\n  font-size: 20px;\n  color: #003c6e;\n  font-weight: 600;\n  padding-left: 30px;\n}\n.profile--address--card .grid--container {\n  display: grid;\n  grid-template-columns: auto auto;\n  padding: 10px;\n}\n.profile--address--card .grid--item {\n  padding: 20px;\n  font-size: 18px;\n  color: #475569;\n  font-weight: 100;\n  text-align: left;\n}\n.profile--address--card .grid--item span {\n  font-size: 14px;\n  font-weight: 600;\n  color: #475569;\n}\n.profile--address--card .grid--item .grid--item--content {\n  display: flex;\n  text-align: left;\n  flex-direction: column;\n  gap: 5px;\n}";
styleInject(css_248z);

// Generated with util/create-component.js
var ProfileAddressCard = function (_a) {
    var country = _a.country, city = _a.city, postCode = _a.postCode, taxId = _a.taxId;
    return (React.createElement("div", { "data-testid": "ProfileCard", className: "profile--address--card" },
        React.createElement("h4", { className: "heading" }, "Address"),
        React.createElement("div", { className: "grid--container" },
            React.createElement("div", { className: "grid--item" },
                React.createElement("div", { className: "grid--item--content" },
                    React.createElement("span", null, "Country"),
                    country)),
            React.createElement("div", { className: "grid--item" },
                React.createElement("div", { className: "grid--item--content" },
                    React.createElement("span", null, "City/State"),
                    city)),
            React.createElement("div", { className: "grid--item" },
                React.createElement("div", { className: "grid--item--content" },
                    React.createElement("span", null, "Postal Code"),
                    postCode)),
            React.createElement("div", { className: "grid--item" },
                React.createElement("div", { className: "grid--item--content" },
                    React.createElement("span", null, "Tax ID"),
                    taxId)))));
};
var ProfilePersonalCard = function (_a) {
    var firstName = _a.firstName, lastName = _a.lastName, email = _a.email, phone = _a.phone;
    return (React.createElement("div", { "data-testid": "ProfileCard", className: "profile--address--card" },
        React.createElement("h4", { className: "heading" }, "Profile"),
        React.createElement("div", { className: "grid--container" },
            React.createElement("div", { className: "grid--item" },
                React.createElement("div", { className: "grid--item--content" },
                    React.createElement("span", null, "First Name"),
                    firstName)),
            React.createElement("div", { className: "grid--item" },
                React.createElement("div", { className: "grid--item--content" },
                    React.createElement("span", null, "Last Name"),
                    lastName)),
            React.createElement("div", { className: "grid--item" },
                React.createElement("div", { className: "grid--item--content" },
                    React.createElement("span", null, "Email Address"),
                    email)),
            React.createElement("div", { className: "grid--item" },
                React.createElement("div", { className: "grid--item--content" },
                    React.createElement("span", null, "Phone"),
                    phone)))));
};

export { Activity, ApplicationTable, Button, CompanyTable, DeleteModal, Entity, EntityTable, LargeLoanCard, MetricRange, PaymentSuccessCard, PersonsTable, PoolCard, ProfileAddressCard, ProfilePersonalCard, ProgressBar, ReceiptTable, RepaymentTable, ResearchTable, ScreeningTable, SignOffTable, SmallLoanCard, StatusWidget };
